app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -126,6 +128,44 @@ const MapImpl: React.FunctionComponent = () => {
             options: {centerMap: true, readOnly: true},
           })
         );
+
+        /*
+        A lot of hardcodeding here. I could probably iterate through the properties later on.
+        */
+        const typeRanges = Object.assign({}, state.typesRanges);
+        for (const feature of featuresInState) {
+          const typeRange = typeRanges[String(feature.type)];
+          if (typeRange) {
+            if (feature.timestamp) {
```

Comment:
This feels like it should all be done in the reducer.

For that matter, calculating featuresInState in the reducer would also be appropriate. If that's the only reason you have this code here, move the featuresInState calculation to the reducer and put it on MapState.